tests/storage/test_roommember.py

```diff
@@ -157,6 +157,55 @@ def test__null_byte_in_display_name_properly_handled(self) -> None:
         # Check that alice's display name is now None
         self.assertEqual(row[0]["display_name"], None)
 
+    def test_room_is_locally_forgotten(self):
+        """Test that when the last local user has forgotten a room it is known as forgotten."""
+        # join two local and one remote user
+        self.room = self.helper.create_room_as(self.u_alice, tok=self.t_alice)
+        self.inject_room_member(self.room, self.u_bob, Membership.JOIN)
+        self.inject_room_member(self.room, self.u_charlie.to_string(), Membership.JOIN)
+        self.assertFalse(
+            self.get_success(self.store.is_locally_forgotten_room(self.room))
+        )
+
+        # local users leave the room and the room is not forgotten
+        self.inject_room_member(self.room, self.u_alice, Membership.LEAVE)
+        self.inject_room_member(self.room, self.u_bob, Membership.LEAVE)
+        self.assertFalse(
+            self.get_success(self.store.is_locally_forgotten_room(self.room))
+        )
+
+        # first user forgets the room, room is not forgotten
+        self.assertNoResult(self.store.forget(self.u_alice, self.room))
```

Comment:
I am need a hint for the failing test:
```
Traceback (most recent call last):
  File "/github/workspace/tests/storage/test_roommember.py", line 178, in test_room_is_locally_forgotten
    self.assertNoResult(self.store.forget(self.u_alice, self.room))
  File "/github/home/.cache/pypoetry/virtualenvs/matrix-synapse-cDQ_M3T8-py3.8/lib/python3.8/site-packages/twisted/trial/_synctest.py", line 777, in assertNoResult
    deferred.addBoth(cb)
builtins.AttributeError: 'coroutine' object has no attribute 'addBoth'

tests.storage.test_roommember.RoomMemberStoreTestCase.test_room_is_locally_forgotten
```

Is the solution?
```python
        forget = ensureDeferred(self.store.forget(self.u_alice, self.room))
        self.assertNoResult(forget)
```

Reply:
`assertNoResult` asserts that a deferred hasn't completed yet. `get_success` and ignoring the result would be more appropriate:

```suggestion
        self.get_success(self.store.forget(self.u_alice, self.room))
```